app/models/form526_submission.rb

```diff
@@ -68,6 +59,41 @@ def start
     start_evss_submission_job
   end
 
+  def start_rrd_job(rrd_processor_class, use_backup_processor: false)
+    workflow_batch = Sidekiq::Batch.new
+    workflow_batch.on(
+      :success,
+      'Form526Submission#rrd_complete_handler',
+      'submission_id' => id
+    )
+    workflow_batch.on(
+      :death,
+      'Form526Submission#rrd_processor_failed_handler',
+      'submission_id' => id,
+      'use_backup_processor' => use_backup_processor
+    )
+    job_ids = workflow_batch.jobs do
+      rrd_processor_class.perform_async(id)
+    end
+    job_ids.first
+  end
+
+  # Called by Sidekiq::Batch as part of the Form 526 submission workflow
+  # When the refactored job fails, this _handler is called to run a backup_processor_class as a job.
+  def rrd_processor_failed_handler(_status, options)
+    submission = Form526Submission.find(options['submission_id'])
+    if options['use_backup_processor']
+      backup_processor_class = submission.rrd_process_selector.processor_class(backup: true)
+    end
+    if backup_processor_class
+      message = "Restarting with backup #{backup_processor_class} for submission #{submission.id}."
+      submission.rrd_process_selector.send_rrd_alert(message)
+      return submission.start_rrd_job(backup_processor_class)
```

Comment:
Good question! Let's see:
* `rrd_complete_handler` returns `submission.start_evss_submission_job`, which returns `job_ids.first`
* `rrd_processor_failed_handler` returns `submission.start_rrd_job` (which returns job_ids.first) or `submission.start_evss_submission_job` (which returns `job_ids.first`)
* `perform_ancillary_jobs_handler` returns `nil` or `submission.perform_ancillary_jobs` (which returns `workflow_batch.jobs` I think)
* `workflow_complete_handler` returns `submission.save` or `Form526SubmissionFailedEmailJob.perform_async`

So there's not consistency on what the `_handler` methods return. For line 91, the main purpose of the `return` is to exit the method so that `submission.start_evss_submission_job` is not executed.